src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AtomicMethods.kt

```diff
@@ -123,7 +123,7 @@ internal fun isAtomicFieldUpdaterMethod(className: String, methodName: String) =
     isAtomicFieldUpdaterClass(className) && methodName in atomicFieldUpdaterMethods
 
 internal fun isVarHandle(obj: Any?) =
-    obj is VarHandle
+    obj != null && generateSequence(obj::class.java) { it.superclass }.any { isVarHandleClass(it.canonicalName ?: "") }
```

Comment:
Could you please add a text explanation here? The code is non-trivial to read.

Reply:
By the way, I think there is a finite list of class inheriting from `VarHandle` (?), so instead of traversing the class hierarchy, we can just check against a set of class name strings. 

You can have a look at these class names here:
https://github.com/JetBrains/lincheck/blob/800f590ef453c3eab5c87b7264c29e580060cec4/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/VarHandleNames.kt#L29